src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -441,40 +457,31 @@ private static void RemoveDocument(ImmutableArray<IIncrementalAnalyzer> analyzer
                         }
                     }
 
-                    private void ResetLogAggregatorIfNeeded(Solution currentSolution)
-                    {
-                        if (currentSolution == null || _processingSolution == null ||
-                            currentSolution.Id == _processingSolution.Id)
-                        {
-                            return;
-                        }
-
-                        SolutionCrawlerLogger.LogIncrementalAnalyzerProcessorStatistics(
-                            this.Processor._registration.CorrelationId, _processingSolution, this.Processor._logAggregator, this.Analyzers);
-
-                        this.Processor.ResetLogAggregator();
-                    }
+                   // this is only used in ResetState to find out solution has changed
+                   // and reset some states
+                    private Solution _lastSolution = null;
 
                     private async Task ResetStatesAsync()
                     {
+                        var currentSolution = this.Processor.CurrentSolution;
+                        var oldSolution = _lastSolution;
+
                         try
                         {
-                            var currentSolution = this.Processor.CurrentSolution;
-
-                            if (currentSolution == _processingSolution)
+                            if (currentSolution == oldSolution)
                             {
                                 return;
                             }
 
-                            // solution has changed
-                            ResetLogAggregatorIfNeeded(currentSolution);
+                            _lastSolution = currentSolution;
 
-                            _processingSolution = currentSolution;
+                            ResetLogAggregatorIfNeeded();
 
                             // synchronize new solution to OOP
                             await currentSolution.Workspace.SynchronizePrimaryWorkspaceAsync(currentSolution, this.CancellationToken).ConfigureAwait(false);
 
-                            await RunAnalyzersAsync(this.Analyzers, currentSolution, (a, s, c) => a.NewSolutionSnapshotAsync(s, c), this.CancellationToken).ConfigureAwait(false);
+                            // always run analyzers with latest solution
+                            await RunAnalyzersAsync(this.Analyzers, this.Processor.CurrentSolution, (a, s, c) => a.NewSolutionSnapshotAsync(s, c), this.CancellationToken).ConfigureAwait(false);
```

Comment:
If the current solution is different than `currentSolution`, would this cause OOP to not receive update or update being delayed?

Reply:
OOP will pull solution when it needs, it just pre-populate OOP with new data so that actual call can be done more quickly. so it doesn't actually matter what solution it is pushing much. but we can change it to CurrentSolution if that is less confusing or add comments.

Reply:
Either way would be fine.